### PR TITLE
feat(agent): deploy stack list module

### PR DESCRIPTION
**Pull request by Dav-14**
## Summary by CodeRabbit

- **Refactor**
	- Enhanced internal event handling for stack operations to improve efficiency and maintainability.

- **New Features**
	- Improved membership listener capabilities for enhanced logic and module management, elevating agent functionality.

- **Tests**
	- Updated tests to align with stack and membership management enhancements, ensuring comprehensive testing coverage.

- **New Functions**
	- Introduced a `Reduce` function to enable operations on slices, enhancing data processing capabilities for improved data manipulation flexibility.

### CODE REVIEW

File: ee/agent/internal/informer_stacks.go [hunk 1]
```diff
@@ -48,28 +56,43 @@ func StacksEventHandler(logger sharedlogging.Logger, membershipClient Membership
...
-			sendStatusFromStack(newStack)
-		},
-		DeleteFunc: func(obj interface{}) {
-			stack := convertUnstructured[*v1beta1.Stack](obj)
-			logger.Infof("Stack '%s' deleted", stack.Name)
-			sendStatus(stack.Name, generated.StackStatus_Deleted)
+				newStack := convertUnstructured[*v1beta1.Stack](newObj)
+
+				if _, ok := stacks[newStack.Name]; !ok {
+					logger.Debugf("Stack '%s' not initialized in memory", newStack.Name)
```
Comment (gfyrag) [inline ee/agent/internal/informer_stacks.go:78]: I'm not sure it will be ok to just ignore the event

**Review by gfyrag — APPROVED** [review]
Just be sure there is no side effect by checking membership status on your dev env please :)

**Review by paul-nicolas — APPROVED** [review]

File: ee/agent/internal/informer_stacks.go (hunk 1 above)
Comment (Dav-14) [reply ee/agent/internal/informer_stacks.go:78]: It's a side effect, where fisrt,
- i was having a stack in progressing while i was ignoring message when !inmemory OR (old,new) was equal
- I'll check again, but this send a status each informer loop,

Comment (Dav-14) [reply ee/agent/internal/informer_stacks.go:78]: That's why i've deleted this check, now it' ok both statuses are set correctly and tested

**Pull request merged**